java/vector/src/main/java/org/apache/arrow/vector/file/json/JsonFileReader.java

```diff
@@ -199,6 +200,9 @@ private void setValueFromParser(ValueVector valueVector, int i) throws IOExcepti
     case FLOAT8:
       ((Float8Vector)valueVector).getMutator().set(i, parser.readValueAs(Double.class));
       break;
+    case VARBINARY:
+      ((VarBinaryVector)valueVector).getMutator().setSafe(i, parser.readValueAs(String.class).getBytes(UTF_8));
+      break;
```

Comment:
a binary value may not be valid UTF-8
do we want to encode in hexadecimal for this?